pybossa/view/account.py

```diff
@@ -1156,30 +1156,37 @@ def add_metadata(name):
     return redirect(url_for('account.profile', name=name))
 
 
-def bookmarks_dict_to_array(bookmarks_dict):
+def bookmarks_dict_to_array(bookmarks_dict, order_by, desc):
+    order_by = "name" if order_by is None else order_by
+    desc = False if desc is None else desc
+
     bookmarks_array = []
     for name, meta in bookmarks_dict.items():
         b = {'name': name}
         b.update(meta)
         bookmarks_array.append(b)
+
+    bookmarks_array.sort(key=lambda b: b[order_by], reverse=desc)
```

Comment:
Not required but in case you want, line 1163 to line 1167 can be re-written as one liner with list comprehension with dict unpacking:
```python
bookmarks_array = [{'name': name, **meta} for name, meta in bookmarks_dict.items()]
```

Reply:
refactored via https://github.com/bloomberg/pybossa/pull/843/commits/81b103681f39a9575aeedfcb5778663d84b2f9df